[Temp] Add operator support for dynamic quant on mobile

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#32128 [Temp] Add operator support for dynamic quant on mobile**
* #32364 QNNPACK: Add support for dynamic requantization.

Summary:

Test Plan:
python test/test_quantized.py TestDynamicQuantizedLinear.test_qlinear_qnnpack

Reviewers:

Subscribers:

Tasks:

Tags:

supriyar committed:
[Temp] Add operator support for dynamic quant on mobile
Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 9d3bc176:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=9d3bc176ea8b8f9ad3af9bee6873c26222c7b3c2).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4224297) pytorch_xla_linux_xenial_py3_6_clang7_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=84412065) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=84412065))
<details>
<summary>
<code>Jan 13 21:36:15 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff</code>
</summary>

```
Jan 13 21:36:15 + XDIR=/var/lib/jenkins/workspace/xla/scripts/.. 
Jan 13 21:36:15 + PTDIR=/var/lib/jenkins/workspace/xla/scripts/../.. 
Jan 13 21:36:15 + python /var/lib/jenkins/workspace/xla/scripts/cond_patch.py /var/lib/jenkins/workspace/xla/scripts/../torch_patches /var/lib/jenkins/workspace/xla/scripts/../.. 
Jan 13 21:36:15 1 out of 1 hunk FAILED 
Jan 13 21:36:15 Applying patch file: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff 
Jan 13 21:36:15 Traceback (most recent call last): 
Jan 13 21:36:15   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 67, in <module> 
Jan 13 21:36:15     patch_repo(args) 
Jan 13 21:36:15   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 49, in patch_repo 
Jan 13 21:36:15     raise RuntimeError('Failed to apply patch: {}'.format(ppath)) 
Jan 13 21:36:15 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff 
Jan 13 21:36:15 + cleanup 
Jan 13 21:36:15 + retcode=1 
Jan 13 21:36:15 + set +x 
Jan 13 21:36:15 =================== sccache compilation log =================== 
Jan 13 21:36:15 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Jan 13 21:36:15 Compile requests                 1 
Jan 13 21:36:15 Compile requests executed        0 
Jan 13 21:36:15 Cache hits                       0 
Jan 13 21:36:15 Cache misses                     0 
Jan 13 21:36:15 Cache timeouts                   0 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

AshkanAliabadi commented on aten/src/ATen/native/quantized/cpu/qnnpack/src/fc-dynamic-run.cc:
```diff
@@ -0,0 +1,119 @@
+      (const void*) ((uintptr_t) packed_w + (nr_block_start + group_index * n_stride) * (k_stride * sizeof(uint8_t) + sizeof(int32_t))),
+      bias,
+      c + (pixel_index + mr_block_start) * c_stride + nr_block_start + group_index * n,
+      c_stride,
```
I think this should be c_stride * sizeof(float).

AshkanAliabadi replied:
OK so I modified the kernels to not need this multiplication.  Please undo these changes that affect the c_stride.

supriyar committed:
Update on "[Temp] Add operator support for dynamic quant on mobile"


Summary:

Test Plan:
python test/test_quantized.py TestDynamicQuantizedLinear.test_qlinear_qnnpack

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[Temp] Add operator support for dynamic quant on mobile"


Summary:

Test Plan:
python test/test_quantized.py TestDynamicQuantizedLinear.test_qlinear_qnnpack

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[Temp] Add operator support for dynamic quant on mobile"


Summary:

Test Plan:
python test/test_quantized.py TestDynamicQuantizedLinear.test_qlinear_qnnpack

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
